operators/pkg/controller/elasticsearch/driver/default.go

```diff
@@ -361,24 +362,39 @@ func (d *defaultDriver) Reconcile(
 	// passed this point, any pods resource listing should check expectations first
 
 	if !esReachable {
-		// We cannot manipulate ES allocation exclude settings if the ES cluster
-		// cannot be reached, hence we cannot delete pods.
-		// Probably it was just created and is not ready yet.
+		// We cannot manipulate ES allocation settings if the ES cluster cannot be reached,
+		// hence we cannot delete pods. Probably the cluster was just created and is not ready yet.
 		// Let's retry in a while.
-		log.Info("ES external service not ready yet for shard migration reconciliation. Requeuing.")
-
+		log.Info("ES external service not ready yet for any HTTP call. Requeuing.")
 		reconcileState.UpdateElasticsearchPending(resourcesState.CurrentPods.Pods())
-
 		return results.WithResult(defaultRequeue)
 	}
 
+	if len(podsState.RunningReady) >= int(es.Spec.NodeCount()) {
+		// Current number of nodes match the expected number of nodes.
+		// There may still be a cluster mutation in progress, but chances are nodes
+		// that were shut down for replacement or restart are back in the cluster.
+		// Make sure shards allocation is enabled, so the current cluster mutation
+		// (if there is one) can progress.
+		log.V(1).Info("Enabling shards allocation")
+		// TODO: optimize by doing the call only if shards allocation is currently disabled
+		ctx, cancel := context.WithTimeout(context.Background(), esclient.DefaultReqTimeout)
+		defer cancel()
+		if err := esClient.EnableShardAllocation(ctx); err != nil {
```

Comment:
The way things work now, we'll do this call at every single reconciliation (hopefully reconciliations don't happen all the time, but only when there are things to reconcile). We actually don't have to do the call if allocations are already enabled. To optimize, see